### PR TITLE
[auth] PruneBonds refactor

### PR DESCRIPTION
**Pull request by romanornr**
Original code checks if `Locktime` isn't expired yet, if so, it performs operations on non-expired bonds
When a bond isn't expired, it stays in its original position in the clients.bonds slice. If a bond is expired, it's added to the `pruned` slice, which creates a need for extra logic to handle the shifting of bonds within `clients.bonds` slice 

But by changing the condition to `if bond.LockTime < lockTimeRefresh` we check if a bond is expired (effectively the logic is reversed), focus on what happens to expired bonds 

Now, when a bond is expired, it's directly added to the `pruned` slice, and the loop continues to the next bond. Basically, getting rid of extra-logic to handle non-expired bonds within the `clients.bonds` slice. 

By handling expired bonds first, we can skip further checks for them and move on to processing non-expired bonds. This logic is also easier to follow. 

For the logging, collect log data first and log it once after the loop

Test passes my side

### CODE REVIEW

**Review by JoeGruffins — APPROVED** [review]

File: server/auth/auth.go [hunk 1]
```diff
@@ -153,22 +153,29 @@ func (client *clientInfo) pruneBonds(lockTimeThresh int64) (pruned []*db.Bond, b
 		return
 	}
 
-	var n int
+	pruned = make([]*db.Bond, 0, len(client.bonds))
```
Comment (buck54321) [inline server/auth/auth.go:156]: We don't really need to pre-allocate `pruned`. It won't be used often, and when it is is will be a single bond pretty much every time.

File: server/auth/auth.go [hunk 2]
```diff
@@ -153,22 +153,29 @@ func (client *clientInfo) pruneBonds(lockTimeThresh int64) (pruned []*db.Bond, b
...
+	expiredBondsInfo := make([]string, 0) // for logging
+	activeCount := 0
```
Comment (buck54321) [inline server/auth/auth.go:158]: I prefer `var n int`.

File: server/auth/auth.go [hunk 3]
```diff
@@ -153,22 +153,29 @@ func (client *clientInfo) pruneBonds(lockTimeThresh int64) (pruned []*db.Bond, b
...
+
 	for _, bond := range client.bonds {
-		if bond.LockTime >= lockTimeThresh { // not expired
-			if len(pruned) > 0 /* n < i */ { // a prior bond was removed, must move this element up in the slice
-				client.bonds[n] = bond
-			}
-			n++
-			bondTier += int64(bond.Strength)
+		if bond.LockTime < lockTimeThresh { // bond is expired
+			expiredBondsInfo = append(expiredBondsInfo, fmt.Sprintf("Expiring user %v bond %v (%s)", client.acct.ID, coinIDString(bond.AssetID, bond.CoinID), dex.BipIDSymbol(bond.AssetID)))
```
Comment (buck54321) [inline server/auth/auth.go:162]: Let's undo this change and just log it here. This doesn't provide any obvious benefit and new allocations and a second loop.

**Review by buck54321 — CHANGES_REQUESTED** [review]

**Review by buck54321 — COMMENTED** [review]
Reversing the lock time logic makes sense, but the other changes don't improve anything, as far as I can tell.

Are you [on Element](https://docs.decred.org/getting-started/joining-matrix-channe)? You can always chat with us in the `DEX Development` room if you're looking for something to work on,.

**Pull request closed without merging**